src/sdg/api/authentication.py

```diff
@@ -16,4 +19,6 @@ def authenticate_credentials(self, key):
             token = model.objects.get(key=key)
             return anon, token
         except model.DoesNotExist:
-            return anon, None  # optional
+            raise exceptions.AuthenticationFailed(
```

Comment:
What if I don't provide a token, doesn't `key` become `None`? If so, there should be an if-statement above to check `if key`

Also, please add a test to check this behaviour.